### DISCUSSION TITLE
The eslint dependency Minimatch causes tsc errors

### DISCUSSION THREAD

**Discussion by tommyasai** [eslint/eslint#17534] [Q&A]
I'm encountering TypeScript errors when I run tsc on my Remix project. The build fails with the following output:

```
$ npm run typecheck --if-present

> typecheck
> tsc

Error: node_modules/@types/glob/index.d.ts(29,42): error TS2[6](https://github.com/tommyasai/blog/actions/runs/6081984724/job/16498805467#step:5:7)94: Namespace '"/home/runner/work/blog/blog/node_modules/minimatch/dist/cjs/index"' has no exported member 'IOptions'.
Error: node_modules/@types/glob/index.d.ts([7](https://github.com/tommyasai/blog/actions/runs/6081984724/job/16498805467#step:5:8)4,30): error TS2724: '"/home/runner/work/blog/blog/node_modules/minimatch/dist/cjs/index"' has no exported member named 'IMinimatch'. Did you mean 'Minimatch'?
Error: Process completed with exit code 2.
```

These errors seem to be related to dependencies that come from ESLint.

## Related Discussion
There's a [StackOverflow discussion](https://stackoverflow.com/questions/75890950/node-modules-minimatch-dist-cjs-index-has-no-exported-member-ioptions) that addresses this issue, but I am not sure how to resolve it in my project.

## Version Information
Node.js: 20.5.1
typescript: ^5.1.6
eslint: ^8.47.0
@types/eslint: ^8.44.2

<details>
<summary>eslintrc.js</summary>

```
/**
 * @type {import('@types/eslint').Linter.BaseConfig}
 */
module.exports = {
  extends: [
    "@remix-run/eslint-config",
    "@remix-run/eslint-config/node",
    "prettier",
  ],
};
```
</details>

Does anyone have any insights on how to resolve this issue? Thank you!

**Comment by Rec0iL99** [comment]
Hi @tommyasai, did you try removing `@types/glob` from your dependencies as the StackOverflow answer suggests?

**Reply by tommyasai** [reply]
@Rec0iL99 
You are right. This was caused by the Remix on my codebase and was resolved with their fix.
https://github.com/remix-run/remix/issues/7295

Thank you for your response. ❤️
